design/gateway-apis-implementation.md

```diff
@@ -62,69 +61,77 @@ Project Contour (the organisation) may investigate a Gateway API based layer 4 s
+Because of this, Contour will watch all events associated with its named GatewayClass, Gateway, HTTPRoute, TLSRoute, and BackendPolicy objects, and filter the objects it takes action on internally.
 
 ### Combining Gateway API with other configuration
 
 When it calculates the Envoy configuration using the DAG, Contour layers different types of configuration in order.
-So, currently, Ingress is overwritten by HTTPProxy, so if an Ingress and a HTTPProxy specify the exact same route, the HTTPProxy will win.
+Currently, Ingress is overwritten by HTTPProxy. 
+If an Ingress and a HTTPProxy specify the exact same route, the HTTPProxy will win.
 
-Once we have the Gateway API available as well, we have to choose the order.
-In this design, I suggest having the order be Ingress is overwritten by HTTPProxy, is overwritten by the Gateway API.
-I could see reversing HTTPProxy and the Gateway API here, but I think this acknowledges that the Gateway API are really an evolution of the ideas in HTTPProxy, and will probably end up being the community standard.
+Once we have the Gateway API available as well, the processing order will be Ingress is overwritten by HTTPProxy, is overwritten by the Gateway API.
+We could potentially reverse HTTPProxy and the Gateway API here, but this acknowledges that the Gateway API are really an evolution of the ideas in HTTPProxy, and will probably end up being the community standard.
 
 ### Status management
 
-Each object in the Gateway API set has its own status definition. In general, Contour will update the status of any object that comes into scope with its details, ensuring that the `observedGeneration` field is set correctly.
+Each object in the Gateway API set has its own status definition. 
+In general, Contour will update the status of any object that comes into scope with its details, ensuring that the `observedGeneration` field is set correctly.
 When objects fall out of scope, any status set by Contour will not be removed.
 It's expected that things that check the status will also check the `observedGeneration` to check if the status information is up-to-date with the current object generation.
 
 ### Per-object design notes
 
 #### GatewayClass
 
-Contour does not watch or interact with GatewayClass in any way.
+Contour watches for GatewayClass resources which can, optionally, define how an instance of Contour should be configured.
+A GatewayClass can reference a configuration resource (Configmap or CRD) which describes how the fleet of Envoys should be provisioned in the cluster along with how they are exposed outside of the cluster.
+
+If the GatewayClass doesn't reference any configuration resources, then a standard set of configuration is utilized much like the Contour QuickStart today.
+
+Contour will need to be configured with the controller name matching the GatewayClass's `Spec.Controller` field.
 
 #### Gateway
 Contour will watch its configured Gateway object.
```

Comment:
Since Contour's not configured with a `Gateway` anymore I'd spell this out a little more - i.e. Contour watches the `Gateway` that uses the `GatewayClass` that has the configured `ControllerName`.

Reply:
Agreed, needs a caveat about confict resolution rules too.

Reply:
There's a note about conflict resolution right after that paragraph.

Reply:
👍